File: web/lab1/src/history/interface/historyManager.ts
import { HistoryEntry } from "../model.ts";

export interface HistoryManager {
  /**
   * Добавляет новую запись в историю
   */
  push(entry: HistoryEntry): void;

  /**
   * Возвращает все записи в истории
   */
  get(): HistoryEntry[];

  /**
   * Возвращает последнее вхождение в историю
   */
  getLastEntry(): HistoryEntry | undefined;

  /**
   * Очищает историю
   */
  cleanHistory(): void;
}

File: web/lab2/src/main/kotlin/efedorov/model/store/HistoryManager.kt
package efedorov.model.store

import efedorov.model.HistoryEntry
import jakarta.servlet.http.HttpSession

interface HistoryManager {
    fun pushEntry(entry: HistoryEntry, session: HttpSession)
    fun getEntries(session: HttpSession): List<HistoryEntry>
    fun getEntiresWithRadius(session: HttpSession, radius: Double): List<HistoryEntry>
}